resurfemg/helper_functions_legacy.py

```diff
@@ -1,1902 +0,0 @@
-
-#     lowest_index = np.argmin(corr_matrix[0][1:])
-#     emg_component = components_tuple[lowest_index]
-
-#     return emg_component
-
-
-# def pick_highest_correlation_array(components_tuple, ecg_lead):
-#     """Here we have a function that takes a tuple with the two parts
-#     of ICA and the array containing the ECG recording, and finds the
-#     ICA component with the highest similarity to the ECG.
-#     Data should not have been finally filtered to envelope level
-
-#     :param components_tuple: tuple of two arrays representing different
-#     signals
-#     :type components_tuple: Tuple[~numpy.ndarray, ~numpy.ndarray]
-
-#     :param ecg_lead: array containing the ECG recording
-#     :type ecg_lead: numpy.ndarray
-
-#     :returns: Array with the highest correlation coefficient
-#      to the ECG lead (should usually be the  ECG)
-#     :rtype: ~numpy.ndarray
-#     """
-#     c0 = components_tuple[0]
-#     c1 = components_tuple[1]
-#     corr_tuple = np.row_stack((ecg_lead, c0, c1))
-#     corr_matrix = abs(np.corrcoef(corr_tuple))
-
-#     # get the component with the highest correlation to ECG
-#     # the matrix is symmetric, so we can check just the first row
-#     # the first coefficient is the autocorrelation of the ECG lead,
-#     # so we can check row 1 and 2
-
-#     hi_index = np.argmax(corr_matrix[0][1:])
-#     ecg_component = components_tuple[hi_index]
-
-#     return ecg_component
-
-
-def working_pipeline_exp(our_chosen_file):
-    """This function is legacy.
-    It produces a filtered respiratory EMG signal from a
-    3 lead sEMG file. A better
-    option is a corresponding function in multi_lead_type
-    The inputs are :code:`our_chosen_file` which we
-    give the function as a string of filename.  The output is the
-    processed EMG signal filtered and seperated from ECG components.
-    The algorithm to pick out the EMG here is by having
-    more peaks.
-
-    :param our_chosen_file: Poly5 file
-    :type our_chosen_file: ~TMSiSDK.file_readers.Poly5Reader
-
-    :returns: final_envelope_a
-    :rtype: ~numpy.ndarray
-    """
-    cut_file_data = bad_end_cutter(
-        our_chosen_file,
-        percent_to_cut=3,
-        tolerance_percent=5,
-    )
-    bd_filtered_file_data = emg_bandpass_butter_sample(
-        cut_file_data,
-        5,
-        450,
-        2048,
-        output='sos',
-    )
-    # end-cutting again to get rid of filtering artifacts
-    re_cut_file_data = bad_end_cutter_for_samples(
-        bd_filtered_file_data,
-        percent_to_cut=3,
-        tolerance_percent=5,
-    )
-    # do ICA
-    components = compute_ICA_two_comp(re_cut_file_data)
-    #  pick components with more peak
-    emg = pick_more_peaks_array(components)
-    # now process it in final steps
-    abs_values = abs(emg)
-    final_envelope_d = emg_highpass_butter(abs_values, 150, 2048)
-    final_envelope_a = naive_rolling_rms(final_envelope_d, 300)
-
-    return final_envelope_a
-
-
```

Comment:
This function (working_pipeline_pre_ml) is not yet introduced in another submodule